workflows/gasclustering.nf

```diff
@@ -47,17 +48,58 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+
+def validateFilePath(String filep){
+    // TODO request clarification if there is a better way to perform this operation
+    if(filep){
+        file_in = file(filep)
+        if(file_in.exists()){
+            return file_in
+        }else{
+            exit 1, "${filep} does not exist but was passed to the pipeline. Exiting now."
+        }
+    }
+    return [] // empty value if file argument is null
+}
```

Comment:
Check the syntax on mine, but off the top of my head you could probably do something like:


```
def validateFilePath(String filep){
    if (filep && file(filep).exists()) {
        return true
    } else {
        return false
    }
}
```

Since the verb of the function is `validate`, if I was calling the function, I would expect it to return `true` or `false`, rather than a file object or an empty list. Also, as the caller, I wouldn't expect an exit in a validate function, so the calling block might look something like:

```

if (!validateFilePath(filep)) {
   exit 1, "${filep} does not exist but was passed to the pipeline. Exiting now."
}

my_file = file(filep)
```

But also make sure you want `file` here and not `path` for Azure compatibility.

Reply:
This makes sense! the intention is partly to allow for optional file inputs. e.g. if null  and emtpy channel is returned and some logic in the module itself can update the arguments to the tool in the process.

I dont know if my description makes sense, so let me know if further clarification is needed.

Reply:
I think the logic (in general) makes sense, but I don't know about throwing an exit in a function like that. I think a useful question is "what is this function going to do?"

Maybe the logic is like this:

- user doesn't provide a path, allow the pipeline to continue (and maybe change the value?)
- user provides a path, path exists, use the provided path
- user provides a path, path doesn't exist, exit pipeline

I wouldn't want an exit in a function that "prepares" or "validates", so the logic probably moves out to the caller:

pseudocode:
```
if path is null:
    change value?
else if path doesn't exist:
    exit

continue program ...
```

```
def my_path = path(the_path)

if (!my_path) {
    my_path = []
} else if (!my_path.exists()) {
    exit 1, "${filep} does not exist but was passed to the pipeline. Exiting now."
```

Reply:
I like this!! I see where your going, I imagine exiting the program is a pretty huge side effect that we would like to avoid. I can move the exit out of the function though and I think that will clarify things.

Reply:
I updated the function to return `null` on error so that the exit is not handled by the function. I think it should address our concnerns?